### PR TITLE
[Filesystem] improve messages on failure

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by jderusse — APPROVED** [review]

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -50,13 +50,13 @@ public function copy(string $originFile, string $targetFile, bool $overwriteNewe
 
         if ($doCopy) {
             // https://bugs.php.net/64634
-            if (false === $source = @fopen($originFile, 'r')) {
-                throw new IOException(sprintf('Failed to copy "%s" to "%s" because source file could not be opened for reading.', $originFile, $targetFile), 0, null, $originFile);
+            if (!$source = self::box('fopen', $originFile, 'r')) {
+                throw new IOException(sprintf('Failed to copy "%s" to "%s" because source file could not be opened for reading: ', $originFile, $targetFile).self::$lastError, 0, null, $originFile);
```
Comment (Nyholm) [inline src/Symfony/Component/Filesystem/Filesystem.php:54]: Why not use the `sprintf` instead of concatenate `self::$lastError`

**Review by Nyholm — COMMENTED** [review]
Thank you.

File: src/Symfony/Component/Filesystem/Filesystem.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Filesystem/Filesystem.php:54]: because fabbot would complain about missing double quotes around, which we don't want

**Review by javiereguiluz — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you Nicolas.

**Pull request merged**